tensorflow_datasets/image/arc.py

```diff
@@ -0,0 +1,126 @@
+_DL_URL = "{}zipball/{}/".format(_BASE_URL, _COMMIT)
+_DL_RESOURCE = tfds.download.Resource(
+    url=_DL_URL,
+    extract_method=tfds.download.ExtractMethod.ZIP)
+_EXTRACT_SUBDIR = "fchollet-ARC-{}".format(_COMMIT[:7])
+
+
+class ARC(tfds.core.GeneratorBasedBuilder):
+  """The Abstraction and Reasoning Corpus (ARC)."""
+
+  VERSION = tfds.core.Version("1.0.0")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        # This is the description that will appear on the datasets page.
+        description=_DESCRIPTION,
+        # tfds.features.FeatureConnectors
+        features=tfds.features.FeaturesDict({
+            # Grids are of varying size
+            "task_id": tfds.features.Text(),
+            "train": tfds.features.Sequence({
+                "input": tfds.features.Sequence(tfds.features.Sequence(tf.int32)),
+                "output": tfds.features.Sequence(tfds.features.Sequence(tf.int32)),
+            }),
+            "test": tfds.features.Sequence({
+                "input": tfds.features.Sequence(tfds.features.Sequence(tf.int32)),
+                "output": tfds.features.Sequence(tfds.features.Sequence(tf.int32)),
+            })
+        }),
+        # If there's a common (input, target) tuple from the features,
+        # specify them here. They'll be used if as_supervised=True in
+        # builder.as_dataset.
+        supervised_keys=None,
+        # Homepage of the dataset for documentation
+        homepage=_BASE_URL,
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Downloads the data, defines the splits and returns SplitGenerators."""
+
+    # dl_manager is a tfds.download.DownloadManager that can be used to
+    # download and extract URLs
+    extracted_dir = dl_manager.download_and_extract(_DL_RESOURCE)
+    data_dir = os.path.join(extracted_dir, _EXTRACT_SUBDIR, "data")
+    train_dir = os.path.join(data_dir, "training")
+    test_dir = os.path.join(data_dir, "evaluation")
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                "directory": train_dir,
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={
+                "directory": test_dir,
+            }),
+    ]
+
+  def _generate_examples(self, directory):
+    """Yields (key, example) tuples from the dataset"""
+    json_filepaths = tf.io.gfile.glob(os.path.join(directory, "*.json"))
+    for json_path in sorted(json_filepaths):
+        try:
+            with tf.io.gfile.GFile(json_path) as f:
+                task = json.load(f)
+            task_id = os.path.basename(json_path)[:-len(".json")]
+            yield task_id, {
+                "task_id": task_id,
+                "train": task["train"],
+                "test": task["test"],
+            }
+        except Exception as ex:
+            logging.exception("Could not load task {}.".format(task_id))
```

Comment:
Why is it required ? Which tasks cannot be loaded ?

It be much more preferable to have an explicit filter for those tasks rather than catch any exception.

Having broad exception catching is bad practice, as it would remove exception which shouldn't have been. (ex: no more disk space, file not found.)

Reply:
The dataset is constantly evolving, and the first versions had some errors (null values). If the user can select the commit they want, then this is not needed anymore, as the user can always pick a commit that works.

Reply:
Sounds good, in this case, please remove the try/except block, and use a commit which works for the default `BuilderConfig()`

Reply:
I removed the exception handling.